applications/minotari_node/src/commands/command/get_peer.rs

```diff
@@ -62,58 +60,54 @@ enum ArgsError {
+            .get_connection(peer_id)
+            .await?
+            .ok_or(ArgsError::NoPeerMatching { original_str })?;
 
-        for peer in peers {
-            let eid = EmojiId::from(&peer.public_key).to_string();
-            println!("Emoji ID: {}", eid);
-            println!("Public Key: {}", peer.public_key);
-            println!("NodeId: {}", peer.node_id);
-            println!("Addresses:");
-            peer.addresses.addresses().iter().for_each(|a| {
-                println!(
-                    "- {} Score: {:?}  - Source: {} Latency: {:?} - Last Seen: {} - Last Failure:{}",
-                    a.address(),
-                    a.quality_score(),
-                    a.source(),
-                    a.avg_latency(),
-                    a.last_seen()
-                        .as_ref()
-                        .map(|t| t.to_string())
-                        .unwrap_or_else(|| "Never".to_string()),
-                    a.last_failed_reason().unwrap_or("None")
-                );
-            });
-            println!("User agent: {}", peer.user_agent);
-            println!("Features: {:?}", peer.features);
-            println!("Flags: {:?}", peer.flags);
-            println!("Supported protocols:");
-            peer.supported_protocols.iter().for_each(|p| {
-                println!("- {}", String::from_utf8_lossy(p));
-            });
-            if let Some(dt) = peer.banned_until() {
-                println!("Banned until {}, reason: {}", dt, peer.banned_reason);
-            }
-            if let Some(dt) = peer.last_seen() {
-                println!("Last seen: {}", dt);
-            }
+        match peer.public_key {
+            Some(pk) => {
+                let pk = pk.try_into_sr25519()?;
+                let eid = EmojiId::from(pk.inner_key()).to_string();
+                println!("Emoji ID: {}", eid);
+                println!("Public Key: {}", pk.inner_key());
+            },
+            None => {
+                println!("Public Key: Unknown");
+            },
+        };
+        println!("NodeId: {}", peer.peer_id);
+        println!("Addresses:");
+        println!(
+            "- {} Latency: {:?}",
+            peer.endpoint.get_remote_address(),
+            peer.ping_latency,
+        );
+        match peer.user_agent {
+            Some(ua) => {
+                println!("User agent: {ua}");
+            },
+            None => {
+                println!("User agent: Unknown");
+            },
         }
+        // TODO: we could also provide this
+        // println!("Supported protocols:");
+        // peer.supported_protocols.iter().for_each(|p| {
+        //     println!("- {}", String::from_utf8_lossy(p));
+        // });
+        // if let Some(dt) = peer.banned_until() {
+        //     println!("Banned until {}, reason: {}", dt, peer.banned_reason);
+        // }
+        // if let Some(dt) = peer.last_seen() {
+        //     println!("Last seen: {}", dt);
+        // }
```

Comment:
```suggestion
         println!("Supported protocols:");
         peer.supported_protocols.iter().for_each(|p| {
             println!("- {}", String::from_utf8_lossy(p));
         });
         if let Some(dt) = peer.banned_until() {
             println!("Banned until {}, reason: {}", dt, peer.banned_reason);
         }
         if let Some(dt) = peer.last_seen() {
             println!("Last seen: {}", dt);
        }
```
this sounds handy to have?

Reply:
Ok I'll implement it

Reply:
We dont persistently store peers (libp2p does not implement this), so get peers is basically getting info about a connected peer